common/socketoptions.cpp

```diff
@@ -7,6 +7,11 @@ using namespace std;
 set<string> true_names = { "1", "yes", "on", "true" };
```

Comment:
None of the types that are used in these constants here throw exceptions on construction. You can check types like `std::set` or `std::basic_string` that they declare `noexcept` for constructor, with condition that underlying type has also noexcept constructor (in C++11 mode).

For logical clarity they indeed need to be constant of course - I'll fix that.

Reply:
OK. But what is the value of having a non-POD globals that is also outside any namespaces? It certainly isn't pretty even if harm is minimal for the moment. It also scares people like me. Can we just move 'true_names' et co to where it is needed?

Reply:
Ok, indeed, can't argue with the fact that this is used currently only in one function (it wasn't always the case), so this can be simply moved to be a static variable in that function. But note that when it will be later found a need to use them anywhere else, they will have to be moved back where they are now.

Reply:
Maybe. Hopefully that will be done within a namespace at that time. Or maybe we can just a namespace today if we really want this variable bouncing around just in case it becomes useful elsewhere in the future.

Reply:
Ops, sorry I was wrong. These `false_names` and `true_names` are used everywhere throughout all applications. So it must stay there, there's no other way. Note also that this part was at best moved to another place because it was there since a long time.

Reply:
OK. Could that be put into a namespace then since that is now part of the library?

Reply:
It is not part of the library and it never was.

Reply:
Ah, I see. I keep thinking of '/common' as part of the lib whereas they those are app helper functions.  Maybe it is worth coming up with a namespace for those too at one point since I presume the intention is to make those helpers available for external apps. But that would better be a separate pull request. Thank you for clarification.

Reply:
No, it wasn't meant to be available for external apps. I was thinking about exposing it for external apps, as lots if things there are pretty useful, but at best we'll do it at some point in the future. The best solution for me would be to create a **real** C++ API for SRT at some point, and make this part something that can be published, for example, as "srtsupport" library. This way the `srt` namespace can be created and both things can be put under this.